### PR TITLE
hearing tab failure handling

### PR DESCRIPTION
**Pull request by SivaprakasamSivaprakasam**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-5444

### Change description ###

Hearing tab error/failure handling

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/cases/containers/case-hearings/case-hearings.component.ts [hunk 1]
```diff
@@ -38,9 +41,18 @@ export class CaseHearingsComponent implements OnInit {
       map(userDetails => userDetails.userInfo.roles)
     );
     this.hearingStore.dispatch(new fromHearingStore.LoadAllHearings(this.caseId));
+    this.hearingState$ = this.hearingStore.pipe(select(fromHearingStore.getHearingsFeatureState));
   }
 
   public ngOnInit(): void {
+    this.sub = this.hearingState$.subscribe(state => {
+      if (state && state.hearingRequest && state.hearingRequest.lastError) {
```
Comment (rayliangatos) [inline src/cases/containers/case-hearings/case-hearings.component.ts:49]: this is the wrong place, it should be the hearingList

File: src/hearings/store/actions/hearing-request.action.ts [hunk 2]
```diff
@@ -34,6 +35,13 @@ export class LoadHearingRequest implements Action {
   }
 }
 
+export class LoadHearingRequestFailure implements Action {
+  public readonly type = LOAD_HEARING_REQUEST_FAILURE
+
+  constructor(public payload: HttpError) {
+  }
+}
```
Comment (rayliangatos) [inline src/hearings/store/actions/hearing-request.action.ts:43]: this is wrong place, it should be in the hearingLIst

File: src/hearings/store/reducers/hearing-request.reducer.ts [hunk 3]
```diff
@@ -79,7 +79,8 @@ export function hearingRequestReducer(currentState = initialHearingRequestState,
         hearingRequestMainModel: updatedHearingRequestMainModel
       };
     }
-    case fromActions.SUBMIT_HEARING_REQUEST_FAILURE: {
+    case fromActions.SUBMIT_HEARING_REQUEST_FAILURE:
+    case fromActions.LOAD_HEARING_REQUEST_FAILURE: {
```
Comment (rayliangatos) [inline src/hearings/store/reducers/hearing-request.reducer.ts:83]: this should be the hearingList

**Review by rayliangatos — CHANGES_REQUESTED** [review]
Logical error, this API is from the getHearingList not get Hearing request.
And we also need to change the mock data to simulate the unhappy path scenario.

File: api/hearings/hearing.mock.ts [hunk 4]
```diff
@@ -42,6 +42,13 @@ export const init = () => {
...
+      // the below exclusion of return 500 just to faciliate testing
+      if (caseIds[0].indexOf('6', caseIds[0].length - 1) > 0) {
+        return [
+          500,
+          null,
+        ];
+      }
       return [
         200,
         EMPTY_HEARINGS_LIST,
```
Comment (rayliangatos) [inline api/hearings/hearing.mock.ts:54]:
```suggestion
      if (caseIds[0].endsWith('6') {
        return [
          500,
          null,
        ];
      }
      return [
        200,
        ...EMPTY_HEARINGS_LIST,
```

File: src/cases/containers/case-hearings/case-hearings.component.html [hunk 5]
```diff
@@ -1,29 +1,40 @@
...
-    [status]="pastAndCancelledStatus"
-    [hearingList$]="pastAndCancelledHearings$"
-    [actions]="hearingsActions">
-  </exui-case-hearings-list>
+<div class="govuk-tabs__panel">
+  <div class="govuk-width-container" *ngIf="serverError">
+    <div class="govuk-main-wrapper">
+        <p class="govuk-body">{{serverError.message}}</p>
+        <span class="govuk-body align-left">
+          <a href="javascript:void(0)" [routerLink]="['/', 'cases', 'case-details', caseId, 'hearings']"
```
Comment (rayliangatos) [inline src/cases/containers/case-hearings/case-hearings.component.html:6]: it seems angular will do nothing to navigate to the same page. so we need to use (click)="reload()"

on the reload() dispatch an action
this.hearingStore.dispatch(new fromHearingStore.LoadAllHearings(this.caseId));

**Review by rayliangatos — APPROVED** [review]
looks good to me, thanks for addressing the comments.

**Review by Kartheek-S — APPROVED** [review]

**Pull request merged**